src/App.tsx

```diff
@@ -1,61 +1,126 @@
+import { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import todosList from './api/todos';
+import users from './api/users';
+
+function getUserById(userId: number) {
+  return users.find(user => user.id === userId) || null;
+}
+
+export const todos = todosList.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState('');
+  const [isCompleted, setIsCompleted] = useState(false);
+  const [listOfTodos, setListOfTodos] = useState(todos);
+  const [userError, setUserError] = useState('');
+  const [titleError, setTitleError] = useState('');
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!todoTitle || todoTitle === '') {
+      setTitleError('Please enter a title');
+
+      return;
+    }
+
+    if (!selectedUser || selectedUser === '0') {
```

Comment:
The condition `selectedUser === '0'` is used to check if a user is selected. Ensure that the `selectedUser` state is consistently initialized and updated to avoid logical errors.